api/app/controllers/spree/api/products_controller.rb

```diff
@@ -56,30 +56,25 @@ def create
         params[:product][:available_on] ||= Time.now
         set_up_shipping_category
 
-        begin
-          @product = Product.new(product_params)
-          if @product.save
-            variants_params.each do |variant_attribute|
-              # make sure the product is assigned before the options=
-              @product.variants.create({ product: @product }.merge(variant_attribute))
-            end
-
-            option_types_params.each do |name|
-              option_type = OptionType.where(name: name).first_or_initialize do |option_type|
-                option_type.presentation = name
-                option_type.save!
-              end
+        @product = Product.new(product_params)
+        if @product.save
+          variants_params.each do |variant_attribute|
+            # make sure the product is assigned before the options=
+            @product.variants.create({ product: @product }.merge(variant_attribute))
+          end
 
-              @product.option_types << option_type unless @product.option_types.include?(option_type)
```

Comment:
this is just how git handles diffs.
the only lines that actually changed from this huge git diff is that I got reid of the begin,rescue block